src/main/java/com/graphhopper/converter/resources/AbstractConverterResource.java

```diff
@@ -67,11 +70,14 @@ boolean isValid(Locale locale) {
         }
     }
 
-    void failIfResponseNotSuccessful(WebTarget target, Status status) {
-        // TODO Maybe limit to == 200?
+    Status failIfResponseNotSuccessful(WebTarget target, Response response) {
+        Status status = new Status(response.getStatus(), response.getStatusInfo().getReasonPhrase());
         if (status.code < 200 || status.code >= 300) {
+            // avoid connection leaks, see #52
+            response.close();
```

Comment:
SO suggests that we have to consume the response, but IMO close is sufficient. WDYT?

Reply:
It looks like closing should be enough. 

The [JavaDoc](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/Response.html#close--) states:
> Close the underlying message entity input stream (if available and open) as well as releases any other resources associated with the response (e.g. buffered message entity data).

Also the [getEntity](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/Response.html#getEntity--) method describes:
> IllegalStateException - if the entity was previously fully consumed as an input stream, or if the response has been closed.

It makes sense that we cannot read the entity of a closed response. So it cannot be consumed anymore. So I think it's safe to assume that it should be released from the ConnectionPool.

I guess it would make sense to first only try to close the response. If it doesn't fix the connection leak, we can still read the entity. But I am pretty sure closing should be enough.